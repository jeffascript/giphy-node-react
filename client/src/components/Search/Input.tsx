import React from 'react';
import { Icon, InputGroup, Tooltip, Whisper } from 'rsuite';
import { VscClearAll } from 'react-icons/vsc';
import ActionButton from '../Button/ActionButton';
import { FlexContainer, StyledInputGroup, StyledInput } from './searchbox.styles';

interface Input {
    state: string;
    handleOnchange: (e: string) => void;
    fetchGifs: (e: any, data: string) => void;
    clearData: () => void;
}

export const Input: React.FC<Input> = ({ state, handleOnchange, fetchGifs, clearData }) => {
    return (
        <>
            <FlexContainer>
                <Whisper
                    trigger="focus"
                    speaker={<Tooltip>{state?.length < 2 ? 'Minimum of 3 letters Required' : null}</Tooltip>}
                    placement={'bottomStart'}
                >
                    <StyledInputGroup inside={true}>
                        <StyledInput
                            onChange={(e) => handleOnchange(e)}
                            value={state}
                            onPressEnter={(e) => fetchGifs(e, state)}
                            size={'lg'}
                            placeholder={'Start typing ... e.g: happy'}
                        />
                        <InputGroup.Button onClick={(e) => fetchGifs(e, state)}>
                            <Icon icon="search" />
                        </InputGroup.Button>
                    </StyledInputGroup>
                </Whisper>
                <ActionButton
                    className="d-flex text-center justify-content-center align-items-center m-auto"
                    onClick={clearData}
                    data-testid="cleardata"
                >
                    <VscClearAll size={22} /> &nbsp; <span>reset</span>
                </ActionButton>
            </FlexContainer>
        </>
    );
};
