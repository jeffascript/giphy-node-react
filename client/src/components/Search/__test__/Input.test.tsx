import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { Input } from '../Input';

const changeHandler = jest.fn();
const clearDataHandler = jest.fn();
const fetchGifsHandler = jest.fn();

const state = 'hello';

test('renders the text input', () => {
    const { rerender, asFragment } = render(
        <Input
            state={state}
            fetchGifs={fetchGifsHandler}
            clearData={clearDataHandler}
            handleOnchange={changeHandler}
        />,
    );

    expect(asFragment()).toMatchSnapshot(`
<DocumentFragment>
    <div
    class="css-3l4gaw"
  >
    <div
      class="rs-input-group css-1fesyhj rs-input-group-inside"
    >
      <input
        class="rs-input css-1rxzna3 rs-input-lg"
        placeholder="Start typing ... e.g: happy"
        type="text"
        value="hello"
      />
      <a
        class="rs-btn rs-btn-default rs-input-group-btn"
      >
        <i
          class="rs-icon rs-icon-search"
        />
        <span
          class="rs-ripple-pond"
        >
          <span
            class="rs-ripple"
          />
        </span>
      </a>
    </div>
    <div
      class="d-flex text-center justify-content-center align-items-center m-auto"
      style="cursor: pointer;"
    >
        <svg
        fill="currentColor"
        height="22"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 12.6l.7.7 1.6-1.6 1.6 1.6.8-.7L13 11l1.7-1.6-.8-.8-1.6 1.7-1.6-1.7-.7.8 1.6 1.6-1.6 1.6zM1 4h14V3H1v1zm0 3h14V6H1v1zm8 2.5V9H1v1h8v-.5zM9 13v-1H1v1h8z"
        />
      </svg>

      <span>
        reset
      </span>
    </div>
  </div>
</DocumentFragment>
`);
    const textInput = screen.getByRole('textbox');
    expect(textInput).toHaveValue(state);
    fireEvent.change(textInput, { target: { value: 'world' } });
    expect(changeHandler).toHaveBeenCalledWith('world');
    rerender(
        <Input
            state={'world'}
            fetchGifs={fetchGifsHandler}
            clearData={clearDataHandler}
            handleOnchange={changeHandler}
        />,
    );
    expect(textInput).toHaveValue('world');
});
