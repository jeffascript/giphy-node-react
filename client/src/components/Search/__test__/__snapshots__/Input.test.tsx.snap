// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the text input: 
<DocumentFragment>
    <div
    class="css-3l4gaw"
  >
    <div
      class="rs-input-group css-1fesyhj rs-input-group-inside"
    >
      <input
        class="rs-input css-1rxzna3 rs-input-lg"
        placeholder="Start typing ... e.g: happy"
        type="text"
        value="hello"
      />
      <a
        class="rs-btn rs-btn-default rs-input-group-btn"
      >
        <i
          class="rs-icon rs-icon-search"
        />
        <span
          class="rs-ripple-pond"
        >
          <span
            class="rs-ripple"
          />
        </span>
      </a>
    </div>
    <div
      class="d-flex text-center justify-content-center align-items-center m-auto"
      style="cursor: pointer;"
    >
        <svg
        fill="currentColor"
        height="22"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 12.6l.7.7 1.6-1.6 1.6 1.6.8-.7L13 11l1.7-1.6-.8-.8-1.6 1.7-1.6-1.7-.7.8 1.6 1.6-1.6 1.6zM1 4h14V3H1v1zm0 3h14V6H1v1zm8 2.5V9H1v1h8v-.5zM9 13v-1H1v1h8z"
        />
      </svg>

      <span>
        reset
      </span>
    </div>
  </div>
</DocumentFragment>
 1`] = `
<DocumentFragment>
  <div
    class="css-3l4gaw"
  >
    <div
      class="rs-input-group css-1fesyhj rs-input-group-inside"
    >
      <input
        class="rs-input css-1rxzna3 rs-input-lg"
        placeholder="Start typing ... e.g: happy"
        type="text"
        value="hello"
      />
      <a
        class="rs-btn rs-btn-default rs-input-group-btn"
      >
        <i
          class="rs-icon rs-icon-search"
        />
        <span
          class="rs-ripple-pond"
        >
          <span
            class="rs-ripple"
          />
        </span>
      </a>
    </div>
    <div
      class="d-flex text-center justify-content-center align-items-center m-auto"
      style="cursor: pointer;"
    >
      <svg
        fill="currentColor"
        height="22"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 12.6l.7.7 1.6-1.6 1.6 1.6.8-.7L13 11l1.7-1.6-.8-.8-1.6 1.7-1.6-1.7-.7.8 1.6 1.6-1.6 1.6zM1 4h14V3H1v1zm0 3h14V6H1v1zm8 2.5V9H1v1h8v-.5zM9 13v-1H1v1h8z"
        />
      </svg>
       Â  
      <span>
        reset
      </span>
    </div>
  </div>
</DocumentFragment>
`;
